### PR TITLE
WFP 1648 more conviction ids removed

### PR DESCRIPTION
**Pull request by markreesmoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UpsertUnallocatedCaseService.kt [hunk 1]
```diff
@@ -52,7 +52,7 @@ class UpsertUnallocatedCaseService(
               unallocatedCaseEntity.crn,
               activeConvictions
             )
-            val caseType = caseTypeEngine.getCaseType(activeConvictions, unallocatedCaseEntity.convictionId)
+            unallocatedCaseEntity.convictionNumber = conviction.convictionNumber
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UpsertUnallocatedCaseService.kt:55]: needed to move this so the convictionNumber is set on new cases before determining the caseType

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/integration/tier/CalculationEventListenerTest.kt [hunk 2]
```diff
@@ -37,25 +36,23 @@ internal class CalculationEventListenerTest : IntegrationTestBase() {
...
-    writeUnallocatedCaseToDatabase(crn, "D0", secondConvictionId, 2)
+    writeUnallocatedCaseToDatabase(crn, "D0", 1)
+    writeUnallocatedCaseToDatabase(crn, "D0", 2)
     publishTierCalculationCompleteMessage(crn)
     checkTierHasBeenUpdated(crn, "B3", 1)
     checkTierHasBeenUpdated(crn, "B3", 2)
   }
 
-  private fun writeUnallocatedCaseToDatabase(crn: String, tier: String, convictionId: Long, convictionNumber: Int) {
+  private fun writeUnallocatedCaseToDatabase(crn: String, tier: String, convictionNumber: Int) {
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/integration/tier/CalculationEventListenerTest.kt:47]: convictionId not relevant here

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/integration/unallocatedcases/GetCaseProbationRecordByCrnTest.kt [hunk 3]
```diff
@@ -144,14 +120,19 @@ class GetCaseProbationRecordByCrnTest : IntegrationTestBase() {
...
+    makeTestRequest(crn, convictionNumber)
       .jsonPath("$.previous[0].offenderManager")
       .doesNotExist()
   }
+
+  private fun makeTestRequest(
+    crn: String,
+    convictionNumber: Int
+  ) = webTestClient.get()
+    .uri("/cases/unallocated/$crn/record/exclude-conviction/$convictionNumber")
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/integration/unallocatedcases/GetCaseProbationRecordByCrnTest.kt:132]: use the new (now only) endpoint for probation record

**Review by carlov20 — APPROVED** [review]
🧹

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UpsertUnallocatedCaseService.kt (hunk 1 above)
Comment (sp-watson) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/UpsertUnallocatedCaseService.kt:55]: Maybe you could have just passed in `conviction.convictionNumber` to `caseTypeEngine.getCaseType`, but no biggie

**Review by sp-watson — APPROVED** [review]

**Pull request merged**